### PR TITLE
*: fix revive linter

### PR DESCRIPTION
**Pull request by fuweid**
Remove old revive_pass in the bash scripts and migirate the revive.toml into golangci linter_settings.

Part of #16610

Please read https://github.com/etcd-io/etcd/blob/main/CONTRIBUTING.md#contribution-flow.

### CODE REVIEW

File: tests/robustness/model/deterministic.go [hunk 1]
```diff
@@ -315,7 +314,7 @@ type MaybeEtcdResponse struct {
 	Error           string
 }
 
-var EtcdFutureRevErr = errors.New("future rev")
+var ErrEtcdFutureRev = errors.New("future rev")
```
Comment (ahrtr) [inline tests/robustness/model/deterministic.go:317]: Did revive linter raise a warning on this?

File: tests/robustness/failpoints.go [hunk 2]
```diff
@@ -421,7 +421,7 @@ func (tb triggerBlackhole) Available(config e2e.EtcdProcessClusterConfig, proces
 	return config.ClusterSize > 1 && process.PeerProxy() != nil
 }
 
-func blackhole(t *testing.T, ctx context.Context, member e2e.EtcdProcess, clus *e2e.EtcdProcessCluster, shouldWaitTillSnapshot bool) error {
+func blackhole(ctx context.Context, t *testing.T, member e2e.EtcdProcess, clus *e2e.EtcdProcessCluster, shouldWaitTillSnapshot bool) error {
```
Comment (ahrtr) [inline tests/robustness/failpoints.go:424]: Did revive linter raise a warning on this?

File: tools/.golangci.yaml [hunk 3]
```diff
@@ -19,14 +19,74 @@ linters:
...
+      - name: indent-error-flow
+        severity: error
+        disabled: false
+      - name: errorf
+        severity: error
+        disabled: false
+      - name: context-keys-type
+        severity: error
+        disabled: false
+      # TODO: enable the following rules
```
Comment (ahrtr) [inline tools/.golangci.yaml:81]: What's the plan for the TODO?

Comment (serathius) [reply tools/.golangci.yaml:81]: Flip `disabled: true`?

**Review by serathius — APPROVED** [review]

File: tools/.golangci.yaml (hunk 3 above)
Comment (ahrtr) [reply tools/.golangci.yaml:81]: I mean why they are disabled now and when to enable them?

It isn't a big deal though.

Comment (serathius) [reply tools/.golangci.yaml:81]: TODO in tests/revive.toml

File: tests/robustness/failpoints.go (hunk 2 above)
Comment (fuweid) [reply tests/robustness/failpoints.go:424]: Yes

```
failpoints.go:424:30: context-as-argument: context.Context should be the first parameter of a function (revive)
func blackhole(t *testing.T, ctx context.Context, member e2e.EtcdProcess, clus *e2e.EtcdProcessCluster, shouldWaitTillSnapshot bool) error {
```

File: tests/robustness/model/deterministic.go (hunk 1 above)
Comment (ahrtr) [reply tests/robustness/model/deterministic.go:317]: thx for the clarification.

File: tests/robustness/failpoints.go (hunk 2 above)
Comment (ahrtr) [reply tests/robustness/failpoints.go:424]: thx

**Review by ahrtr — APPROVED** [review]

File: tools/.golangci.yaml (hunk 3 above)
Comment (fuweid) [reply tools/.golangci.yaml:81]: @ahrtr The TODO was defined in the tests/revive.toml. I can file issue to track this~

**Pull request merged**